### ISSUE TITLE
Application crashes when no declaration of R.color.colorPrimary not R.attr.colorPrimary

### ISSUE THREAD

**Issue by keima** [yshrsmz/LicenseAdapter#43]
### Version: 

2.0.2

### What went wrong?:

Crash app. 

<details>

```
2018-05-22 12:41:06.223 26057-26057/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: net.yslibrary.licenseadapter.example, PID: 26057
    android.content.res.Resources$NotFoundException: File  from drawable resource ID #0x7f04002c: .xml extension required
        at android.content.res.ResourcesImpl.loadComplexColorForCookie(ResourcesImpl.java:985)
        at android.content.res.ResourcesImpl.loadComplexColorFromName(ResourcesImpl.java:806)
        at android.content.res.ResourcesImpl.loadColorStateList(ResourcesImpl.java:885)
        at android.content.res.Resources.getColor(Resources.java:1800)
        at android.content.Context.getColor(Context.java:508)
        at android.support.v4.content.ContextCompat.getColor(ContextCompat.java:418)
        at net.yslibrary.licenseadapter.internal.ViewHolderBase.<init>(ViewHolderBase.java:16)
        at net.yslibrary.licenseadapter.internal.LibraryViewHolder.<init>(LibraryViewHolder.java:0)
        at net.yslibrary.licenseadapter.LicenseAdapter.onCreateViewHolder(LicenseAdapter.java:78)
        at net.yslibrary.licenseadapter.LicenseAdapter.onCreateViewHolder(LicenseAdapter.java:31)
        at android.support.v7.widget.RecyclerView$Adapter.createViewHolder(RecyclerView.java:6685)
        at android.support.v7.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:5869)
        at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5752)
        at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5748)
        at android.support.v7.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2232)
        at android.support.v7.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1559)
        at android.support.v7.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1519)
        at android.support.v7.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:614)
        at android.support.v7.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:3812)
        at android.support.v7.widget.RecyclerView.dispatchLayout(RecyclerView.java:3529)
        at android.support.v7.widget.RecyclerView.onLayout(RecyclerView.java:4082)
        at android.view.View.layout(View.java:18804)
        at android.view.ViewGroup.layout(ViewGroup.java:5951)
        at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1079)
        at android.view.View.layout(View.java:18804)
        at android.view.ViewGroup.layout(ViewGroup.java:5951)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
        at android.view.View.layout(View.java:18804)
        at android.view.ViewGroup.layout(ViewGroup.java:5951)
        at android.support.v7.widget.ActionBarOverlayLayout.onLayout(ActionBarOverlayLayout.java:443)
        at android.view.View.layout(View.java:18804)
        at android.view.ViewGroup.layout(ViewGroup.java:5951)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
        at android.view.View.layout(View.java:18804)
        at android.view.ViewGroup.layout(ViewGroup.java:5951)
        at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1741)
        at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1585)
        at android.widget.LinearLayout.onLayout(LinearLayout.java:1494)
        at android.view.View.layout(View.java:18804)
        at android.view.ViewGroup.layout(ViewGroup.java:5951)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
        at com.android.internal.policy.DecorView.onLayout(DecorView.java:891)
        at android.view.View.layout(View.java:18804)
        at android.view.ViewGroup.layout(ViewGroup.java:5951)
        at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2659)
        at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2375)
        at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1526)
        at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7102)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:927)
        at android.view.Choreographer.doCallbacks(Choreographer.java:702)
2018-05-22 12:41:06.223 26057-26057/? E/AndroidRuntime:     at android.view.Choreographer.doFrame(Choreographer.java:638)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:913)
        at android.os.Handler.handleCallback(Handler.java:751)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6682)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1520)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1410)
```

</details>

### Steps to reproduce the problem:

In `example` module of LicenseAdapter, change below files:

colors.xml:
```colors.xml
<?xml version="1.0" encoding="utf-8"?>
<resources>
  <!--<color name="colorPrimary">#3F51B5</color>-->
  <!--<color name="colorPrimaryDark">#303F9F</color>-->
  <!--<color name="colorAccent">#910a38</color>-->

  <color name="primaryColor">#3F51B5</color>
  <color name="primaryColorDark">#303F9F</color>
  <color name="accentColor">#910a38</color>
</resources>
```

styles.xml
```xml
<resources>

  <!-- Base application theme. -->
  <style name="AppTheme" parent="Theme.AppCompat.Light.DarkActionBar">
    <!-- Customize your theme here. -->
    <item name="colorPrimary">@color/primaryColor</item>
    <item name="colorPrimaryDark">@color/primaryColorDark</item>
    <item name="colorAccent">@color/accentColor</item>
  </style>

</resources>
```

### Cause of a problem (suspected)

[In ViewHolderBase.java](https://github.com/yshrsmz/LicenseAdapter/blob/9db21b2c1c1bc49b4eea371c465c7ab1f27d0eb5/library/src/main/java/net/yslibrary/licenseadapter/internal/ViewHolderBase.java#L16), It refer to `R.color.colorPrimary` not `attr/colorPrimary`.
And `ids.xml`, `colorPrimary` is defined as color resource so `R.color.colorPrimary` is **required** value but it's not seen other libs.

( Maybe ViewHolderBase is retrieve primaryColor from `R.attr.primaryColor`. [solution](https://stackoverflow.com/a/17277714/1161994) but I'm not tried yet :( )

**Comment by SUPERCILEX** [comment]
Are you by chance using the AGP `v3.2.0-alpha15`? With a `com.android.dynamic-feature` module? I've encountered [this issue](https://issuetracker.google.com/issues/79997225) too and it's not the repo fault—resource linking is broken with dynamic features unfortunately.

**Comment by keima** [comment]
I'm using AGP 3.0.1 and not use dynamic-feature (as it is this repo's`example` module).
(... but issue you suggested is interesting, thanks)

I use Android Studio 3.2-canary15 until just now, but problem is reproduced switch back to AS 3.1.2 stable.

**Comment by SUPERCILEX** [comment]
Yeah, but check your `build.gradle` file... the Android team does a poor job of explaining that Android Studio has nothing to do with how your code is built. It's all delegated to the Gradle plugin and if you're using the latest canary, chances are it forced you to upgrade your Android Gradle Plugin too.

**Comment by keima** [comment]
In AS3.1.2, 
no diff except colors.xml and styles.xml (checked by `git status`), 
remove cache ( `./gradlew clean`, remove `.gradle` dir ), 
then run `example` app, but app is crash. sadly :(

---

To explain again, the root of this issue is **requred by** `R.color.colorPrimary`, not `R.attr.colorPrimary`.
This library expected that [CustomTab toolbar color](https://github.com/yshrsmz/LicenseAdapter/blob/9db21b2c1c1bc49b4eea371c465c7ab1f27d0eb5/library/src/main/java/net/yslibrary/licenseadapter/internal/ViewHolderBase.java#L21) is `?attr/colorPrimary` (isn't it right?).
But actual is `LibraryViewHolder` referenced `R.color.colorPrimary` directly.

If my app's styles.xml defined `colorPrimary` as `@color/material_blue_500`, works great expect LicenseAdapter (anyway app crash when showing LicensesAdapter and defined no `R.color.colorPrimary`).
This behavior is undocumented.
And this is unexpected for me that `R.color.colorPrimary` is required. And required definitions named by`colorPrimary` is not phrased actual behavior. e.g. `la_customTabsToolbarColor` is not bad (wait, I don't say "good". ;-) ).

Additional problem is...
If we want to change CustomTab toolbar to different by `R.color.colorPrimary`, it is impossible at current implementation.

I think best approach is create own theme like `preferenceTheme` at `preference-v14`.
I want to implement and send PR, but I have no time now...

**Comment by yshrsmz** [comment]
Hi, I've just released v2.1.0 with the suggested fix.
Please try it out and let me know if you have any issue.
Thanks for reporting!
